src/routes/SearchRoutes/searchApi.ts

```diff
@@ -0,0 +1,43 @@
+import express, { Router, Request, Response, NextFunction } from "express";
+import User, { UserModel } from "../../Database/Models/User";
+import UserController from "../../Database/DbController/UserController";
+import Errors from "../../Errors/Errors";
+import { Request2 } from "../../Authentication/auth";
+import GoogleAPI from "../../Integrations/GoogleBooks";
+import BookController from "../../Database/DbController/BookController";
+import GoogleSearch from "../../Integrations/GoogleSearch";
+const googleApi = new GoogleAPI();
+const googleEngine = new GoogleSearch()
+const bookController = new BookController();
+const router: Router = express.Router();
+
+router.get("/", async function (req: Request2, res: Response) {
+  try {
+    const { q } = req.query;
+    if (q) {
+      const books = await googleApi.searchBooks({ keyword: q.toString() });
+      bookController.addBulkBook(books);
```

Comment:
> bookController.addBulkBook(books);

You should add an `await` here (addBulkBook is async)

Reply:
but I'm not waiting for the results of the call, do I have to add await, I figured why let the user wait for this to happen, let me know what you think?